docs/design/automate-skus.md

```diff
@@ -0,0 +1,124 @@
+# Azure VM SKU and Locations Automation
+
+This document proposes a design to automate the updating of some data structures used to represent Azure constants in AKS Engine.
+
+## Problem
+
+AKS Engine represents some information about Azure in code constants or similar read-only data structures. Specifically, Azure data center location names and Virtual Machine SKU details are needed for validation and for enabling certain VM features.
+
+To stay in sync with Azure, these data structures must be refreshed periodically. To date, that has been a task for maintainers to perform on-demand. AKS Engine has often lagged behind Azure's introduction of new locations or VM SKUs.
+
+A python script will regenerate some, but not all of the data structures.
+
+## Goals
+
+- On a regular schedule, notify the AKS Engine team if new locations or SKUs exist
+- Automate the creation of the pull request with the new locations or SKUs
+
+## Non-goals
+
+## Proposal: scriptable aks-engine commands
```

Comment:
I love that this proposal also standardizes the way we deliver this information to users.